plugins/cryptokiplugin/plugin.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2017 Jolla Ltd.
+ * Contact: Chris Adams <chris.adams@jollamobile.com>
+ * All rights reserved.
+ * BSD 3-Clause License, see LICENSE.
+ *
+ * This plugin is aimed to provide a high level interface to interact
+ * with Cryto Token USB devices supported PKSC#11 standard.
+ *
+ * Copyright (C) 2017 Open Mobile Platform LLC.
+ * Contact: Denis Semakin <d.semakin@omprussia.ru>
+ * All rights reserved.
+ */
+
+#ifndef CRYPTOKI_PLUGIN_H
+#define CRYPTOKI_PLUGIN_H
+
+#include "Crypto/extensionplugins.h"
+
+#include <QObject>
+#include <QByteArray>
+#include <QCryptographicHash>
+
+namespace Sailfish {
+
+namespace Crypto {
+
+namespace Daemon {
+
+namespace Plugins {
+
+class Q_DECL_EXPORT CryptokiPlugin : public Sailfish::Crypto::CryptoPlugin
+{
+    Q_OBJECT
+    Q_PLUGIN_METADATA(IID Sailfish_Crypto_CryptoPlugin_IID)
+    Q_INTERFACES(Sailfish::Crypto::CryptoPlugin)
+
+public:
+    CryptokiPlugin(QObject *parent = Q_NULLPTR);
+    ~CryptokiPlugin();
+
+    bool isTestPlugin() const Q_DECL_OVERRIDE {
+#ifdef SAILFISH_CRYPTO_BUILD_TEST_PLUGIN
+        return true;
+#else
+        return false;
+#endif
+    }
+
+    QString name() const Q_DECL_OVERRIDE { return QLatin1String("org.sailfishos.crypto.plugin.cryptoki"); }
+    bool canStoreKeys() const Q_DECL_OVERRIDE { return false; }
+
+    Sailfish::Crypto::CryptoPlugin::EncryptionType encryptionType() const Q_DECL_OVERRIDE { return Sailfish::Crypto::CryptoPlugin::SoftwareEncryption; }
```

Comment:
May be SecurePeripheralEncryption instead of SoftwareEncryption?

Reply:
Yeap